modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectRelationshipLocalServiceImpl.java

```diff
@@ -482,6 +482,18 @@ private void _validate(
 			throw new ObjectRelationshipTypeException("Invalid type " + type);
 		}
 
+		ObjectDefinition objectDefinition =
+			_objectDefinitionPersistence.fetchByPrimaryKey(objectDefinitionId1);
+
+		if (objectDefinition.isSystem() &&
+			!Objects.equals(
+				type, ObjectRelationshipConstants.TYPE_ONE_TO_MANY)) {
+
+			throw new ObjectRelationshipTypeException(
+				"Invalid type for system object definition " +
+					objectDefinitionId1);
+		}
+
```

Comment:
I'm not sure if we need this. I am able to create a One to One relationship via API since there is no validation in the service layer, for example, although this feature is still under development.